pkg/yqutil/yqutil_test.go

```diff
@@ -0,0 +1,65 @@
+package yqutil
+
+import (
+	"testing"
+
+	"gotest.tools/v3/assert"
+)
+
+func TestEvaluateExpressionSimple(t *testing.T) {
+	expression := `.cpus = 2 | .memory = "2GiB"`
+	content := `
+# CPUs
+cpus: null
+
+# Memory size
+memory: null
+`
+	// Note: yq currently removes empty lines, but not comments
+	expected := `
+# CPUs
+cpus: 2
+# Memory size
+memory: 2GiB
+`
+	out, err := EvaluateExpression(expression, []byte(content))
+	assert.NilError(t, err)
+	assert.Equal(t, expected, string(out))
+}
+
+func TestEvaluateExpressionComplex(t *testing.T) {
+	expression := `.mounts += {"location": "foo", "mountPoint": "bar"}`
+	content := `
+# Expose host directories to the guest, the mount point might be accessible from all UIDs in the guest
+# 🟢 Builtin default: null (Mount nothing)
+# 🔵 This file: Mount the home as read-only, /tmp/lima as writable
+mounts:
+- location: "~"
+  # Configure the mountPoint inside the guest.
+  # 🟢 Builtin default: value of location
+  mountPoint: null
+`
+	// Note: yq will use canonical yaml, with indented sequences
```

Comment:
Not sure what "canonical YAML" means, but I would not consider indented sequences to be "canonical"; they are just the result of simplistic generators... 😛

Reply:
Yeah, in this case I think the issues listed (blank lines, indents) are quirks of the particular yaml library used.

But there were no good config or workaround for it, people were resorting to diff/patch and other such hacks

Reply:
I think it is ok; just having comments preserved is already great.